components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
 
 - (void)setText:(NSString *)text {
   self.textLabel.text = text;
+  self.isLayoutDirty = YES;
 }
 
 - (NSString *)text {
   return self.textLabel.text;
 }
 
-- (void)setImage:(UIImage *)image {
-  if (image) {
-    _image = image;
-    if (!self.iconImageViewContainer) {
-      CGRect iconImageContainerFrame =
-          CGRectMake(0, 0, kIconImageContainerSideLength, kIconImageContainerSideLength);
-      UIView *iconImageViewContainer = [[UIView alloc] initWithFrame:iconImageContainerFrame];
-      UIImageView *iconImageView = [[UIImageView alloc] initWithImage:image];
-      CGRect iconImageFrame = CGRectMake(0, 0, kIconImageSideLength, kIconImageSideLength);
-      iconImageView.frame = iconImageFrame;
-      iconImageView.contentMode = UIViewContentModeScaleAspectFit;
-      iconImageView.clipsToBounds = YES;
-      [iconImageViewContainer addSubview:iconImageView];
-      self.iconImageViewContainer = iconImageViewContainer;
-      self.iconImageView = iconImageView;
-      [self addSubview:iconImageViewContainer];
-    } else {
-      self.iconImageView.image = image;
-    }
+- (void)setIcon:(UIImage *)icon {
+  if (icon == nil) {
+    [self.iconImageView removeFromSuperview];
+    self.iconImageView = nil;
+    self.isLayoutDirty = YES;
+  } else if (self.iconImageView == nil) {
+    CGRect iconImageViewFrame =
+        CGRectMake(0, 0, kIconImageViewSideLength, kIconImageViewSideLength);
+    UIImageView *iconImageView = [[UIImageView alloc] initWithFrame:iconImageViewFrame];
+    iconImageView.image = icon;
+    iconImageView.translatesAutoresizingMaskIntoConstraints = NO;
+
+    NSLayoutConstraint *iconImageWidthConstraint =
+        [NSLayoutConstraint constraintWithItem:iconImageView
+                                     attribute:NSLayoutAttributeWidth
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:nil
+                                     attribute:NSLayoutAttributeNotAnAttribute
+                                    multiplier:1
+                                      constant:kIconImageViewSideLength];
+
+    NSLayoutConstraint *iconImageHeightConstraint =
+        [NSLayoutConstraint constraintWithItem:iconImageView
+                                     attribute:NSLayoutAttributeHeight
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:nil
+                                     attribute:NSLayoutAttributeNotAnAttribute
+                                    multiplier:1
+                                      constant:kIconImageViewSideLength];
+
+    [NSLayoutConstraint
+        activateConstraints:@[ iconImageWidthConstraint, iconImageHeightConstraint ]];
+
+    iconImageView.contentMode = UIViewContentModeCenter;
+    iconImageView.clipsToBounds = YES;
+    self.iconImageView = iconImageView;
+    [self.containerView addSubview:self.iconImageView];
+    self.isLayoutDirty = YES;
```

Comment:
This can be removed and instead we can have `[self invalidateIntrinsicContentSize]`

Reply:
Same reason as mentioned above in https://github.com/material-components/material-components-ios/pull/6830#discussion_r264361551. Marking as resolved to keep the conversation in a single thread.